plutus-core/plutus-ir/src/PlutusIR/Transform/Inline.hs

```diff
@@ -144,29 +183,38 @@ processTerm
     :: Inlining tyname name uni fun a m
     => Term tyname name uni fun a
     -> m (Term tyname name uni fun a)
-processTerm = \case
-    v@(Var _ n) -> do
+processTerm = handleTerm <=< traverseOf termSubtypes applySubstitution where
+    handleTerm = \case
+        v@(Var _ n) -> do
+            subst <- get
```

Comment:
pull this block out into `substFreshenedTerm` to make the parallel with `substFreshenedType` obvious?